### PR TITLE
Issue 30 large sequence numbers

### PR DESCRIPTION
**Pull request by aljones15**
Addresses: https://github.com/digitalbazaar/bedrock-edv-storage/issues/30

Adds tests for

```
should insert a document with sequence number of 0
should insert a document with sequence number of 1
should insert a document with sequence number of 2**31-1
should insert a document with sequence number of 2**31
should insert a document with sequence number of 2**31+1
should insert a document with sequence number of 2**32-1
should insert a document with sequence number of 2**32
should insert a document with sequence number of 2**32+1
should insert a document with sequence number of middle betwixt 2**32-1 and MAX_SAFE_INTEGER
should insert a document with sequence number of random betwixt 2**31-1 and MAX_SAFE_INTEGER
should insert a document with sequence number of MAX_SAFE_INTEGER-1
should insert a document with sequence number of MAX_SAFE_INTEGER
should fail to insert a document with a negative sequence number
```

### CODE REVIEW

File: test/mocha/helpers.js [hunk 1]
```diff
@@ -22,11 +24,31 @@ const {Ed25519Signature2018, RsaSignature2018} = suites;
...
 exports.KMS_MODULE = 'ssm-v1';
 // algorithm required for the jwe headers
 exports.JWE_ALG = 'ECDH-ES+A256KW';
 
+const largeRange = Number.MAX_SAFE_INTEGER - Math.pow(2, 32);
+
+// gets a number between MAX_SAFE_INTEGER & 2^32.
+exports.largeNumber = () => {
+  // if Math.random is 1 the number will be 2^32 otherwise it will be larger.
+  return Number.MAX_SAFE_INTEGER - Math.floor(largeRange * Math.random());
```
Comment (mattcollier) [inline test/mocha/helpers.js:39]: -1 to using randomness here as we will have no idea what actually got tested or what works.

@dlongley will have to say exactly what the need is here, but I would suggest some fixed/deterministic series of test that test min/middle/max values that are supposed to work.

Comment (aljones15) [reply test/mocha/helpers.js:39]: we do know what got tested a number larger than `Math.pow(2, 32)` but smaller than `Number.MAX_SAFE_INTEGER`. This also tests all numbers in that range albeit slowly @davidlehn recommended.

Comment (davidlehn) [reply test/mocha/helpers.js:39]: I agree it should at least test a static set of values around the limits as appropriate.  Something like 2^32-1, 2^32, 2^32+1, (MAX-2^32)/2, MAX-1, MAX.  And maybe toss in some error cases like -1 (and 0?).  I'm assuming logic handling this doesn't know about BigInts yet?

Comment (aljones15) [reply test/mocha/helpers.js:39]: I'm willing to add the tests for min `Math.pow(2,32)` middle, and max `Number.MAX_SAFE_INTEGER` asked for provided we still allow the one I already wrote as I do think fuzz testing this functionality has value in the long run and it's not a good idea to waste work. So let me know. I can add the tests in a moment.
I also like @davidlehn idea of adding tests for negative sequence numbers to ensure it errors when that happens although that was not part of the issue.

@mattcollier  @davidlehn do we have a deal? thumbs up if so.

**Comment by mattcollier** [comment]
@aljones15 CI/coverage stuff just hit master, please rebase.

File: test/mocha/helpers.js [hunk 2]
```diff
@@ -22,11 +24,50 @@ const {Ed25519Signature2018, RsaSignature2018} = suites;
...
 const {Ed25519KeyPair} = require('crypto-ld');
 
+const getRandomBytes = promisify(crypto.randomBytes);
+
 // for key generation
 exports.KMS_MODULE = 'ssm-v1';
 // algorithm required for the jwe headers
 exports.JWE_ALG = 'ECDH-ES+A256KW';
 
+const min64Bit = Math.pow(2, 32);
```
Comment (davidlehn) [inline test/mocha/helpers.js:34]: - This naming here and in test names is confusing.  "min64Bit" seems like it should be 0 or -2^63 (depending if signed).
- JS range is -(2^53 - 1) to 2^53 - 1, so best to avoid saying 64b.

Comment (aljones15) [reply test/mocha/helpers.js:34]: would minLarheNumber be better? Any idea what the semantic meaning on the number ranges @dlongley suggested are? What does 2^32 mean? I googled it and it is slightly larger than the upper limitof a 32-bit number so I was assuming this was a 64 vs 32 bit number issue. Perhaps `minBigInt` would be better?

Comment (dlongley) [reply test/mocha/helpers.js:34]: Should use `0xFFFFFFFF` and can name it `max32Bit` since that's what it is.

Comment (davidlehn) [reply test/mocha/helpers.js:34]: This is getting confusing to even talk about. Perhaps code in https://github.com/digitalbazaar/bedrock-edv-storage/pull/32 is easier to understand?

Comment (aljones15) [reply test/mocha/helpers.js]: With @davidlehn 's changes in are we happy here?

File: test/mocha/helpers.js (hunk 1 above)
Comment (aljones15) [reply test/mocha/helpers.js]: Assuming this can be marked resolved taken @davidlehn 's changes? My apologies about the wait.
